dbus_next/proxy_object.py

```diff
@@ -213,20 +284,12 @@ def get_interface(self, name: str) -> BaseProxyInterface:
         if name in self._interfaces:
             return self._interfaces[name]
 
-        try:
-            intr_interface = next(i for i in self.introspection.interfaces if i.name == name)
-        except StopIteration:
+        intr_interface = self.introspection.get_interface(name)
+        if intr_interface is None:
             raise InterfaceNotFoundError(f'interface not found on this object: {name}')
 
         interface = self.ProxyInterface(self.bus_name, self.path, intr_interface, self.bus)
 
-        for intr_method in intr_interface.methods:
-            interface._add_method(intr_method)
-        for intr_property in intr_interface.properties:
-            interface._add_property(intr_property)
-        for intr_signal in intr_interface.signals:
-            interface._add_signal(intr_signal, interface)
```

Comment:
I wanted to get rid of the need for these loops which extract and load the interface methods/properties/signals into the proxy object and rather allow `BaseProxyInterface` to look them up directly from the interface object. With some further refactoring I think this should ultimately make it possible to do full state tracking of a hierarchical object tree(ie auto-binding to appropriate `InterfacesAdded`, `InterfacesRemoved` and `PropertiesChanged` signals) to give an improved [objectmanager](https://dbus.freedesktop.org/doc/dbus-specification.html#standard-interfaces-objectmanager) style interface with the ability to do tree partial updates.